### PR TITLE
Do not panic on redis connection failure; just log and rely on reconn…

### PR DESCRIPTION
**Pull request by maniksurtani**
…ects

R: @mmarkus

### CODE REVIEW

File: buckets/redis/bucket.go [hunk 1]
```diff
@@ -74,7 +74,13 @@ func (bf *bucketFactory) Init(cfg *config.ServiceConfig) {
 func (bf *bucketFactory) connectToRedis() {
 	// Set up connection to Redis
 	bf.client = redis.NewClient(bf.redisOpts)
-	logging.Printf("Connection established. Time on Redis server: %v", time.Unix(toInt64(bf.client.Time().Val()[0], 0), 0))
+	redisResults := bf.client.Time().Val()
+	if len(redisResults) == 0 {
+		logging.Fatalf("Cannot connect to Redis. TIME returned %v", redisResults)
```
Comment (zhao-squareup) [inline buckets/redis/bucket.go]: `Fatalf` will kill the process as well. Is it intended?

**Comment by mmarkus** [comment]
LGTM!

File: buckets/redis/bucket.go (hunk 1 above)
Comment (maniksurtani) [inline buckets/redis/bucket.go]: That's a good point; no.

Comment (maniksurtani) [inline buckets/redis/bucket.go]: Updated.

**Pull request merged**